authentication/AuthenticationExample/Utility/DataSourceProvider/DataSourceProvider.swift

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+
+
+class DataSourceProvider<Datasource: DataSourceProvidable>: NSObject, UITableViewDataSource, UITableViewDelegate {
+    
+    weak var delegate: DataSourceProviderDelegate?
+    
+    private var emptyView: UIView?
+    
+    private var sections: [Datasource.Section]!
+    
+    convenience init(dataSource: [Datasource.Section]?, emptyStateView: UIView? = nil, tableView: UITableView? = nil) {
```

Comment:
The idea is that when we load our User view and the there is no user signed in, we still create a  `DataSourceProvider` object like: `DataSourceProvider(user?.sections, ..)`. Since the user is null though, `user?.sections` is null and if thats the case then we just default to the default `[Datasource.section]()`

Reply:
Ah gotcha - thanks.